metricbeat/mb/lightmetricset.go

```diff
@@ -83,7 +86,18 @@ func (m *LightMetricSet) Registration(r *Register) (MetricSetRegistration, error
 		// At this point host parser was already run, we need to run this again
 		// with the overriden defaults
 		if registration.HostParser != nil {
-			base.hostData, err = registration.HostParser(base.module, base.host)
+			u, err := url.Parse(base.hostData.URI)
```

Comment:
Take into account that there can be hosts that don't parse as urls, like some formats for MySQL or PostgreSQL, or in general any metricset that uses a custom host parser, or the `PassThruHostParser`.

At this point of the code we only want to parse hosts with the `HostParser`, and we should parse the original setting in the configuration, doesn't `base.host` contain it? :thinking:

Reply:
Well, all tests passed here unless I missed something. What I want to do I fix the issue while using `http` module as base one, so if you know any different workaround, it would be great.

@jsoriano do you suggest to fallback to base.host in case of err here (line 89)?

Reply:
> Well, all tests passed here unless I missed something.

Yep, I guess that there are no tests for light modules with configured hosts that cannot be parsed by `url.Parse` :slightly_frowning_face: Each metricset can implement its own host parser, for their own host strings. We shouldn't assume that they are URLs in all cases.

> What I want to do I fix the issue while using http module as base one, so if you know any different workaround, it would be great.

Yes, I see the problem, actually this call to `HostParser` in this override for light modules was also motivated by some http service. In that moment the problem was that the `query` defined in a manifest was not being considered, because this is added in the `URLHostParserBuilder` (same parser as the one used in `http` module).

I don't know of any workaround at the moment.

> @jsoriano do you suggest to fallback to base.host in case of err here (line 89)?

I mean that in the call to `HostParser` we should use the same host (or an equivalent one) to the one in the configuration file. I was wondering if `base.host` would contain it, but I see now that `base.host` is modified in the first call to the `HostParser` in `metricbeat/mb/builders.go`.

Maybe we can try using `base.hostData.URI`, the URI there should be the complete host.
```
base.hostData, err = registration.HostParser(base.module, base.hostData.URI)
```
But I am not sure if all hosts parsers actually maintain the complete host.

Other possible source to get the original source is `base.module.Config().Hosts`, but this can contain all the hosts configured in a module and we may don't know which one is the one for the current metricset.

Maybe we need to keep the original host in some other field in the `BaseMetricSet` (apart from `host` that gets modified by some host parsers).

Reply:
As discussed offline, let's do this change, but keeping previous behaviour for hosts that don't parse as URLs. We will think further in other possible cases where this host parser override won't work as expected.

Reply:
@jsoriano Thanks, fixed!